### PR TITLE
[T5A2][W15-B4]Jasper Lee

### PR DESCRIPTION
**Pull request by jasperlee27**

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -39,7 +42,12 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     /**
      * Executes the command and returns the result.
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws Exception;
+
+    /**
+     * Return true if command mutates data.
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java]: try to be more specific on this header comment; the abstract class is a contract that we are providing to the client classes, hence the documentation should be precise. e.g., think what happens when you try to add a duplicate entry to address book. does it still mutate data?

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -78,18 +82,25 @@ public CommandResult execute(String userCommandText) throws Exception {
...
-     * @throws Exception if there was any problem during command execution.
+     * @throws Exception
+     *             if there was any problem during command execution.
      */
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
+            storage.save(addressBook);
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java]: may be you need to check how `LogicTest` works now? does it fail?

**Comment by okkhoy** [comment]
@jasperlee27 
Some comments added. Please ack & close the PR after reading comments.

**Comment by jasperlee27** [comment]
Acknowledged

**Pull request closed without merging**